catalog/app/containers/Bucket/requests.js

```diff
@@ -1235,3 +1235,13 @@ export const objectAccessCounts = ({ s3, analyticsBucket, bucket, path, today })
     today,
     window: 365,
   })
+
+export const ensurePackageIsPresent = async ({ s3, bucket, name }) => {
+  const listOfRevisions = await bucketListing({
```

Comment:
i think you should just use `s3.listObjectV2` directly, bc you dont need all the things happening in the `bucketListing`, just `listObjectV2({ Bucket, Prefix, MaxKeys: 1 })` and then check `.Contents.KeyCount`

Reply:
done